ibis/backends/sql/dialects.py

```diff
@@ -27,6 +28,32 @@
     sge.ArraySort: rename_func("arraySort"),
     sge.LogicalAnd: rename_func("min"),
     sge.LogicalOr: rename_func("max"),
+    sge.DateFromParts: lambda self, e: sg.func(
```

Comment:
I actually find reading these in the sqlglot dialect harder to understand than the previous version in the compiler. These are pretty complicated implementations (length-wise), doing more in the dialect means we have additional places to check for complex logic when trying to understand how something is compiled (compiler, rewrite rules, our `dialects.py` file, sqlglot's dialects, sqlglot's rewrite rules). If possible, I'd find it easier to understand compilation if we try to keep most of our implemented logic in the compiler itself rather than the dialects here.

Just a -0.25 though, if you feel strongly about doing it this way then that's fine.

Reply:
Unfortunately without these (or something like it), it's a non-starter to implement this without using a new operation. Happy to try the no-new-op route, but to do that we need sqlglot to correctly compile `DateFromParts` correctly for all the backends it can, which requires porting the previous code to this.

Reply:
In general I agree with your point, but this happened to be an annoying case where it seemed better to centralize this logic in sqlglot rather than repeating in the compiler.